### ISSUE TITLE
ESP32: Some pins are not properly configured

### ISSUE THREAD

**Issue by SergioGasquez** [esp-rs/esp-hal#594]
See [errata 3.6](https://www.espressif.com/sites/default/files/documentation/esp32_errata_en.pdf), we currently have a fix for this (https://github.com/esp-rs/esp-hal/blob/f315d2bf09f059d543aaac6783cee79217aa612d/esp-hal-common/src/soc/esp32/gpio.rs#L575-L658) but is not always called, e.g. the i2c constructor does not apply this workaround.

**Comment by bjoernQ** [comment]
Even with ESP-IDF (e.g. with a changed i2c_simple_main.c example) I don't get I2C to work on these pins without external pull-ups

**Comment by liebman** [comment]
This even fails for simple gpio input with pull up (at least with `gpio33`).  Using the following leaving `gpio33` unconnected and it waits for `gpio33` top go high - it should be pulled up but it keeps waiting.

```rust
#![no_std]
#![no_main]

use esp_backtrace as _;
use esp_println::println;
use hal::{clock::ClockControl, peripherals::Peripherals, prelude::*, Delay, IO};

#[entry]
fn main() -> ! {
    let peripherals = Peripherals::take();
    let system = peripherals.SYSTEM.split();

    let clocks = ClockControl::max(system.clock_control).freeze();

    let io = IO::new(peripherals.GPIO, peripherals.IO_MUX);
    let input = io.pins.gpio33.into_pull_up_input();

    let mut delay = Delay::new(&clocks);

    loop {
        println!("wait for high...");
        while input.is_low().unwrap() {
            delay.delay_ms(10u32);
        }
        println!("wait for low...");
        while input.is_high().unwrap() {
            delay.delay_ms(10u32);
        }
    }
}
```

**Comment by liebman** [comment]
I found that for `gpio32` and `gpio33` they were reversed in the `errata36` function.  I'll send a PR.
Based on the datasheet:
gpio32 = x32p
gpio33 = x32n

**Comment by liebman** [comment]
PR added #1053

**Comment by SergioGasquez** [comment]
Closed by #1053
